django/thunderstore/repository/models/thread.py

```diff
@@ -0,0 +1,29 @@
+from django.db import models
+from ulid2 import generate_ulid_as_uuid
+
+
+class Thread(models.Model):
+    uuid = models.UUIDField(
+        default=generate_ulid_as_uuid,
+        editable=False,
+        unique=True,
+        primary_key=True,
+    )
+
+
+class CommentsThreadMixin(models.Model):
+    """Mixin used to add comments to a model."""
+
+    comments_thread = models.OneToOneField(
+        Thread,
+        on_delete=models.SET_NULL,
+        related_name="parent",
+        null=True,
+    )
+
+    class Meta:
+        abstract = True
+
+    def delete(self, *args, **kwargs):
```

Comment:
This should probably use `@transaction.atomic`, wouldn't want the object deleting without the thread deleting.

Reply:
Is it possible that this would create a transaction after something is already deleted, for example from another mixin?